miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt

```diff
@@ -97,9 +129,20 @@ internal class RealMiniApp(
         }
     }
 
+    private suspend fun setManifestFromApi(appId: String, versionId: String) {
+        val apiManifest = getMiniAppManifest(appId, versionId)
+        val cachedManifest = downloadedManifestCache.readDownloadedManifest(appId)
+
+        if (cachedManifest != apiManifest)
```

Comment:
This seems a little strange to compare the manifests directly. What we actually want to compare is the `versionId` for the downloaded manifest and then only download the new manifest if the cached manifest is for a different `versionId`. This will avoid doing an extra network request in the case that the mini app is already downloaded.

Could you update the `DownloadedManifestCache` to also store the `versionId`? I guess you would need to wrap the manifest in another object, similar to what iOS did here: https://github.com/rakutentech/ios-miniapp/pull/227/files#diff-e2aeedd67de0916c8e00eba89a3fcd7d60f5d49c2a466eb995eed0624f90d5f9R59

Or maybe there's another way to achieve this?

Reply:
I have added a separate class called `CachedManifest` for this.